File: problem/Maximum_Subarray/maxSubArray.c
int max(int a, int b)
{
    return (a > b) ? a : b;
}

int maxSubArray(int *nums, int numsSize)
{
    int res = nums[0];
    int total = 0;

    for (int i = 0; i < numsSize; i++)
    {
        if (total < 0)
        {
            total = 0;
        }
        total += nums[i];
        res = max(res, total);
    }
    return res;
}

File: problem/Maximum_Subarray/maxSubArray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let res = nums[0];
    let total = 0;

    for (let n of nums) {
        if (total < 0) {
            total = 0;
        }

        total += n;
        res = Math.max(res, total);
    }
    
    return res;    
};